textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcher.java

```diff
@@ -91,91 +104,148 @@ public void open() throws Exception {
+     *        and scans the documents for matches Presently 2 types of
+     *        KeywordOperatorType are supported:
+     * 
+     *        SourceOperatorType.SCANOPERATOR:
+     * 
+     *        Loop through the dictionary entries. For each dictionary entry,
+     *        loop through the tuples in the operator. For each tuple, loop
+     *        through the fields in the attributelist. For each field, loop
+     *        through all the matches. Returns only one tuple per document. If
+     *        there are multiple matches, all spans are included in a list. Java
+     *        Regex is used to match word boundaries.
+     * 
+     *        Ex: If dictionary word is "Lin", and text is
+     *        "Lin is Angelina's friend", matches should include Lin but not
+     *        Angelina.
+     * 
+     *        SourceOperatorType.KEYWORDOPERATOR:
+     * 
+     *        Loop through the dictionary entries. For each dictionary entry,
+     *        keywordmatcher's getNextTuple is called using
+     *        KeyWordOperator.BASIC. Updates span information at the end of the
+     *        tuple.
+     * 
+     *        SourceOperatorType.PHRASEOPERATOR:
+     * 
+     *        Loop through the dictionary entries. For each dictionary entry,
+     *        keywordmatcher's getNextTuple is called using
+     *        KeyWordOperator.PHRASE. The span returned is the span information
+     *        provided by the keywordmatcher's phrase operator.
      * 
-     * @overview Loop through the dictionary entries. For each dictionary entry,
-     *           loop through the tuples in the operator. For each tuple, loop
-     *           through all the fields. For each field, loop through all the
-     *           matches. Returns only one tuple per document. If there are
-     *           multiple matches, all spans are included in a list. Java Regex
-     *           is used to match word boundaries. Ex : If text is
-     *           "Lin is Angelina's friend" and the dictionary word is "Lin",
-     *           matches should include Lin but not Angelina.
      */
     @Override
     public ITuple getNextTuple() throws Exception {
-        if (attributeIndex < searchInAttributes.size()) {
-            IField dataField = dataTuple.getField(searchInAttributes.get(attributeIndex).getFieldName());
+        if (predicate.getSourceOperatorType() == DataConstants.SourceOperatorType.PHRASEOPERATOR) {
+            if ((dataTuple = sourceOperator.getNextTuple()) != null) {
+                return dataTuple;
+            } else if ((dictionaryValue = predicate.getNextDictionaryValue()) != null) {
+                KeywordPredicate keywordPredicate = new KeywordPredicate(dictionaryValue, predicate.getAttributeList(),
+                        KeywordOperatorType.PHRASE, predicate.getAnalyzer(), predicate.getDataStore());
+                sourceOperator = new KeywordMatcher(keywordPredicate);

```

Comment:
My understanding of the logic here is, after KeywordMatcher finishes the previous dictionary entry, we get a new KeywordMatcher with the next entry.
So I'm wondering if we should close the previous KeywordMatcher by calling `sourceOperator.close()`?
Same comment applies to [line 244](https://github.com/TextDB/textdb/pull/113/files#diff-372b6458873675337faca8b5828b3877R244)